bodhi/server/util.py

```diff
@@ -278,17 +351,44 @@ def sanity_check_repodata(myurl):
     h.setopt(librepo.LRO_URLS, [myurl])
     h.setopt(librepo.LRO_LOCAL, True)
     h.setopt(librepo.LRO_CHECKSUM, True)
+    h.setopt(librepo.LRO_IGNOREMISSING, False)
+    r = librepo.Result()
     try:
-        h.perform()
+        h.perform(r)
     except librepo.LibrepoException as e:
         rc, msg, general_msg = e.args
         raise RepodataException(msg)
 
-    updateinfo = os.path.join(myurl, 'updateinfo.xml.gz')
-    if os.path.exists(updateinfo):
-        ret = subprocess.call(['zgrep', '<id/>', updateinfo])
-        if not ret:
-            raise RepodataException('updateinfo.xml.gz contains empty ID tags')
+    repo_info = r.getinfo(librepo.LRR_YUM_REPO)
+    primary_sack = hawkey.Sack()
+    hk_repo = hawkey.Repo(myurl)
+    try:
+        hk_repo.filelists_fn = repo_info['filelists']
+        hk_repo.presto_fn = repo_info['prestodelta']
+        hk_repo.primary_fn = repo_info['primary']
+        hk_repo.repomd_fn = repo_info['repomd']
+        hk_repo.updateinfo_fn = repo_info['updateinfo']
+    except KeyError:
+        raise RepodataException('Required part not in repomd.xml')
```

Comment:
I manually ran the CI tests on my laptop, and they failed due to missing coverage on this exception handler.

Reply:
I have a test to address this now. I'll push it up shortly:

```
diff --git a/bodhi/tests/server/test_util.py b/bodhi/tests/server/test_util.py
index 206ec88f..d916816b 100644
--- a/bodhi/tests/server/test_util.py
+++ b/bodhi/tests/server/test_util.py
@@ -16,6 +16,7 @@
 # You should have received a copy of the GNU General Public License
 # along with this program; if not, write to the Free Software
 # Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+from xml.etree import ElementTree
 import json
 import os
 import shutil
@@ -577,6 +578,24 @@ class TestSanityCheckRepodata(unittest.TestCase):

         self.assertEqual(str(exc.exception), 'Comps file empty')

+    def test_repomd_missing_updateinfo(self):
+        """If the updateinfo data tag is missing in repomd.xml, an Exception should be raised."""
+        util.mkmetadatadir(self.tempdir)
+        repomd_path = os.path.join(self.tempdir, 'repodata', 'repomd.xml')
+        repomd = ElementTree.parse(repomd_path)
+        ElementTree.register_namespace('', 'http://linux.duke.edu/metadata/repo')
+        root = repomd.getroot()
+        # Find the <data type="updateinfo"> tag and delete it
+        for data in root.findall('{http://linux.duke.edu/metadata/repo}data'):
+            if data.attrib['type'] == 'updateinfo':
+                root.remove(data)
+        repomd.write(repomd_path, encoding='UTF-8', xml_declaration=True)
+
+        with self.assertRaises(util.RepodataException) as exc:
+            util.sanity_check_repodata(self.tempdir)
+
+        self.assertEqual(str(exc.exception), 'Required part not in repomd.xml')
+

 class TestType2Icon(unittest.TestCase):
     """Test the type2icon() function."""
```